File: common/src/main/java/com/thesilentnights/commands/common/Email.java
package com.thesilentnights.commands.common;

import com.mojang.brigadier.arguments.StringArgumentType;
import com.mojang.brigadier.builder.LiteralArgumentBuilder;
import com.thesilentnights.service.EmailService;
import net.minecraft.commands.CommandSourceStack;
import net.minecraft.commands.Commands;

public class Email implements CommonCommands {
    @Override
    public LiteralArgumentBuilder<CommandSourceStack> getCommand() {
        return Commands.literal("email")
                .then(Commands.literal("bind")
                        .then(Commands.argument("email", StringArgumentType.string())
                                .then(Commands.argument("emailConfirm", StringArgumentType.string())
                                        .executes(commandContext -> EmailService.bindEmail(commandContext)?1:0))));
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/commands/server/common/Email.kt
package com.thesilentnights.easylogin.commands.server.common

import com.mojang.brigadier.arguments.StringArgumentType
import com.mojang.brigadier.builder.LiteralArgumentBuilder
import com.mojang.brigadier.context.CommandContext
import com.thesilentnights.easylogin.service.EmailService
import net.minecraft.commands.CommandSourceStack
import net.minecraft.commands.Commands

class Email : CommonCommands {
    override val command: LiteralArgumentBuilder<CommandSourceStack>
        get() = Commands.literal("email")
            .then(
                Commands.literal("bind")
                    .then(
                        Commands.argument(
                            "email",
                            StringArgumentType.greedyString()
                        )
                            .executes { commandContext: CommandContext<CommandSourceStack> ->
                                //TODO email
                                if (EmailService.bindEmail(commandContext)) {
                                    1
                                } else {
                                    0
                                }
                            }
                    )
            )
}
